framework/di/Container.php

```diff
@@ -665,7 +665,21 @@ public function resolveCallableDependencies(callable $callback, $params = [])
 
             if (PHP_VERSION_ID >= 80000) {
                 $class = $param->getType();
-                $isClass = $class !== null && !$param->getType()->isBuiltin();
+                if ($class instanceof \ReflectionUnionType) {
+                    $isClass = false;
+                    foreach ($class->getTypes() as $type) {
+                        if (!$type->isBuiltin()) {
+                            $class = $type;
+                            $isClass = true;
+                            break;
+                        }
+                    }
+                } elseif ($class instanceof \ReflectionIntersectionType) {
+                    throw new InvalidConfigException("Unable to resolve intersection types yet"); // TODO: decide strategy how to
```

Comment:
I think we can do it the same as union types i.e. taking first not built-in one.

Reply:
I made this version first, but decided that it will cause errors in cases when resolved type does not fit all intersection types. I'll change it on monday